src/components/StatusBadge.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <Badge
+    :class="[backgroundClass, textClass]"
+    class="capitalize"
+  >
+    {{ statusLocale }}
+  </Badge>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed } from 'vue'
+import { ProgramState } from '/~/types/index'
+import { useI18n } from 'vue-i18n'
+
+export default defineComponent({
+  props: {
+    light: {
+      type: Boolean,
+      default: false,
+    },
+    status: {
```

Comment:
If the status can only take some known values, we could add a TypeScript annotation as well as a runtime validated.